yarl/_url.py

```diff
@@ -1160,9 +1160,7 @@ def join(self, url):
 
         if self.path[-1] == "/":
             # using an intermediate to avoid URL.joinpath dropping query & fragment
-            parts["path"] = self._normalize_path(
-                URL(self.path).joinpath(other.path).path
-            )
+            parts["path"] = self._normalize_path(self._make_child([other.path]).path)
         else:
             # …
             # and relativizing ".."
```

Comment:
Can we use make child for this branch as well?

Reply:
I was wrong here.
But as I just saw you've figured this out yourself already.